### ISSUE TITLE
Valid negate rule ignored in does_negate_rule when matched against ignore rule with double-star (**)

### ISSUE THREAD

**Issue by vsevolodp** [libgit2/libgit2#3393]
libgit2 version: master

Valid negate rule ignored in does_negate_rule.

.gitignore content
```python
code/projects/**/packages/*
!code/projects/**/packages/repositories.config # this rule is ignored
```
I think this is because wrong check for "If we're dealing with a directory" in does_negate_rule:
```C
if (!strchr(rule->pattern, '*'))
	error = git_buf_printf(&buf, "%s/*", rule->pattern);
else
	error = git_buf_puts(&buf, rule->pattern);
```
Changing condition like below fixes problem:
```C
if (rule->flags & GIT_ATTR_FNMATCH_LEADINGDIR && !(rule->flags & GIT_ATTR_FNMATCH_NEGATIVE))
	error = git_buf_printf(&buf, "%s/*", rule->pattern);
else
	error = git_buf_puts(&buf, rule->pattern);
```

**Comment by ethomson** [comment]
Note: it sounds like this is a regression from 0.21.1

**Comment by vsevolodp** [comment]
To be precise it was implemented in e05b2ff127e57eaed9900ddd7b9ac199d43268f8

**Comment by carlosmn** [comment]
Fixed via #3394
